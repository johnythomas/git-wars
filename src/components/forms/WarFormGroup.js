import React, { Component } from "react";
import { Grid } from "semantic-ui-react";
import WarForm from "./WarForm";
import { getUser } from "../../utils/api";
import User from "../User";

class WarFormGroup extends Component {
  state = {
    user1: null,
    user2: null
  };

  onSubmit = user => username => {
    getUser(username).then(usr =>
      this.setState(() => ({
        [user]: usr
      }))
    );
  };

  render() {
    return (
      <Grid columns={2} divided>
        <Grid.Row>
          <Grid.Column>
            {this.state.user1 ? (
              <User user={this.state.user1} />
            ) : (
              <WarForm onSubmit={this.onSubmit("user1")} />
            )}
          </Grid.Column>
          <Grid.Column>
            {this.state.user2 ? (
              <User user={this.state.user2} />
            ) : (
              <WarForm onSubmit={this.onSubmit("user2")} />
            )}
          </Grid.Column>
        </Grid.Row>
      </Grid>
    );
  }
}

export default WarFormGroup;
