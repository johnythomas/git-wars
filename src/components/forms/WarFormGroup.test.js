import React from "react";
import { shallow } from "enzyme";
import axios from "axios";
import MockAdapter from "axios-mock-adapter";
import WarFormGroup from "./WarFormGroup";

describe("WarFormGroup", () => {
  const id = "superman";
  const user = {
    name: "John Doe",
    avatar_url: "http://github.com/johnDoe",
    location: "US",
    followers: 100,
    following: 10
  };

  beforeEach(() => {
    const mockAdapter = new MockAdapter(axios);
    mockAdapter.onGet(`https://api.github.com/users/${id}`).reply(200, user);
  });

  it("should render correctly", () => {
    const wrapper = shallow(<WarFormGroup />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should have 2 WarForms", () => {
    const wrapper = shallow(<WarFormGroup />);
    expect(wrapper.find("WarForm")).toHaveLength(2);
  });

  it("should change the state when onSubmit is called", async () => {
    const wrapper = shallow(<WarFormGroup />);
    await wrapper.instance().onSubmit("user1")(id);
    expect(wrapper.state().user1).toEqual(user);
  });

  it("should update the UI after fetching the user details", async () => {
    const wrapper = shallow(<WarFormGroup />);
    await wrapper.instance().onSubmit("user1")(id);
    await wrapper.instance().onSubmit("user2")(id);
    wrapper.update();
    expect(wrapper.find("User")).toHaveLength(2);
  });
});
