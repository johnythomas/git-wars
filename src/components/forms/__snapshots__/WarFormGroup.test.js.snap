// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WarFormGroup should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WarFormGroup />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <GridRow>
        <GridColumn>
                <WarForm
                        onSubmit={[Function]}
                />
        </GridColumn>
        <GridColumn>
                <WarForm
                        onSubmit={[Function]}
                />
        </GridColumn>
</GridRow>,
      "columns": 2,
      "divided": true,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <GridColumn>
            <WarForm
                        onSubmit={[Function]}
            />
</GridColumn>,
          <GridColumn>
            <WarForm
                        onSubmit={[Function]}
            />
</GridColumn>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <WarForm
              onSubmit={[Function]}
/>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "onSubmit": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <WarForm
              onSubmit={[Function]}
/>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "onSubmit": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <GridRow>
          <GridColumn>
                    <WarForm
                              onSubmit={[Function]}
                    />
          </GridColumn>
          <GridColumn>
                    <WarForm
                              onSubmit={[Function]}
                    />
          </GridColumn>
</GridRow>,
        "columns": 2,
        "divided": true,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <GridColumn>
              <WarForm
                            onSubmit={[Function]}
              />
</GridColumn>,
            <GridColumn>
              <WarForm
                            onSubmit={[Function]}
              />
</GridColumn>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <WarForm
                onSubmit={[Function]}
/>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "onSubmit": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <WarForm
                onSubmit={[Function]}
/>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "onSubmit": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
